src/topology/algebra/uniform_ring.lean

```diff
@@ -150,8 +156,61 @@ instance : comm_ring (completion R) :=
       (assume a b, by rw [← coe_mul, ← coe_mul, mul_comm]),
  ..completion.ring }
 
-end uniform_space.completion
+@[simp] lemma map_mul_eq_mul_coe (r : R) :
+  completion.map ((*) r) = (*) (r : completion R) :=
+begin
+  ext x,
+  refine completion.induction_on x _ (λ s, _),
+  { exact is_closed_eq (completion.continuous_map) (continuous_mul_left (r : completion R)) },
+  { rw [map_coe (uniform_continuous_mul_const R r), coe_mul] },
+end
+
+instance : algebra R (completion R) :=
+{ commutes' := λ r x, by simp only [ring_hom.to_fun_eq_coe, mul_comm],
+  smul_def' := λ r x, congr_fun (map_mul_eq_mul_coe R r) x,
+  .. (uniform_space.completion.coe_ring_hom : R →+* completion R) }
 
+lemma algebra_smul_eq (r : R) (x : (completion R)) :
+  r • x = (r : completion R) * x :=
+by rw algebra.smul_def; refl
```

Comment:
This lemma doesn't look very useful to me; a better lemma would be that `algebra_map _ _ x = coe x`